library/src/visualize/shape.rs

```diff
@@ -557,7 +671,15 @@ fn layout(
             let outset = outset.relative_to(frame.size());
             let size = frame.size() + outset.sum_by_axis();
             let pos = Point::new(-outset.left, -outset.top);
-            let shape = ellipse(size, fill, stroke.left);
+            let shape: Shape;
+            match kind {
+              ShapeKind::Ngon(vertices) => {
+                shape = ngon(vertices, size, fill, stroke.left);
+              }
+              _ => {
+                shape = ellipse(size, fill, stroke.left);
+              }
+            }
```

Comment:
Without getting into the discussion regarding the validity of this approach, here's a quick suggestion.

```suggestion
            let shape = match kind {
              ShapeKind::Ngon(vertices) => {
                ngon(vertices, size, fill, stroke.left)
              }
              _ => ellipse(size, fill, stroke.left)
            };
```